llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -3867,6 +3905,10 @@ SDValue DAGCombiner::visitSUB(SDNode *N) {
   if (SDValue V = combineFixedwidthToAVGCEILU(N, DAG))
     return V;
 
+  // Try to match AVGCEILS fixedwidth pattern
+  if (SDValue V = combineFixedwidthToAVGCEILS(N, DAG))
```

Comment:
Merge this into combineFixedwidthToAVGCEILU (rename to combineFixedwidthToAVGCEIL)